src/main/scala/org/wikimedia/analytics/hdfstools/HdfsRsyncExec.scala

```diff
@@ -594,15 +594,27 @@ class HdfsRsyncExec(config: HdfsRsyncConfig) {
     }
 
     /**
-     * Function deleting an existing dst path directory if it is empty
+     * Function deleting an existing dst path directory if it is empty.
+     * Only logs in case of dry-run mode
      *
      * @param dstOpt the optional dst directory path
+     * @return true if the folder has been deleted for real, false if no deletion or dry-run mode
      */
-    private def pruneEmptyAsNeeded(dstOpt: Option[Path]): Unit = {
-        dstOpt.foreach(dst => {
-            if (config.pruneEmptyDirs && config.dstFs.listStatus(dst).isEmpty) {
-                log.debug(s"PRUNE_DIR - $dst")
-                config.dstFs.delete(dst, true)
+    private def pruneEmptyAsNeeded(dstOpt: Option[Path]): Boolean = {
+        dstOpt.exists(dst => {
+            if (config.pruneEmptyDirs) {
+                if (config.dryRun) {
+                    log.info(s"PRUNE_DIR_IF_EMPTY [dryrun]- $dst")
+                    false
+                } else if (config.dstFs.listStatus(dst).isEmpty) {
+                    log.debug(s"PRUNE_DIR - $dst")
+                    config.dstFs.delete(dst, true)
+                    true
+                } else {
+                    false
+                }
+            } else {
+                false
```

Comment:
Huh, I added a comment about restructuring this!  It's gone!

I don't love all the return paths here.  Can we simplify? Something like

```scala
var pruned = false
if (config.pruneEmptyDirs) {
  if (config.dryRun) {
     log.info(...)
  } else if (...isEmpty) {
    log.debug(...)
    ...delete(...)
    pruned = true
 }
}

pruned

Reply:
Done :)